### PR TITLE
feat: rectangles on pdf

### PR DESCRIPTION
**Pull request by pritesh177**
- used context to store the common state of pdf doc throughout the application.
- added overlayCanvas layer over the canvas layer to highlight areas we want.
- component **RectangleDrawer** enables us to draw the rectangle.

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import type { Metadata } from "next";
 import localFont from "next/font/local";
 import "./globals.css";
+import { PdfDocProvider } from "./context/PdfDocContext";
```
Comment (otaku-tekkurai) [inline app/layout.tsx:4]: convert this to alias

File: app/page.tsx [hunk 2]
```diff
@@ -1,61 +1,97 @@
...
-    };
-    await page.render(renderContext).promise;
-  }else{
-    console.error("Falied to get canvas context");
-  }
-  }
+  };
+
+  const renderPdf = async (pdfDoc: PDFDocumentProxy, pageNumber: number) => {
+    if (canvasRef.current && overlayCanvasRef.current) {
```
Comment (otaku-tekkurai) [inline app/page.tsx:39]: do we need this ?

File: app/page.tsx [hunk 3]
```diff
@@ -1,61 +1,97 @@
...
+      const page = await pdfDoc.getPage(pageNumber);
+      const viewport = page.getViewport({ scale: 1.5 }); 
+
+      const canvas = document.getElementById("pdfCanvas") as HTMLCanvasElement;
+      const overlayCanvas = overlayCanvasRef.current;
```
Comment (otaku-tekkurai) [inline app/page.tsx:44]: remove this or make sure you maintain consistency

File: app/page.tsx [hunk 4]
```diff
@@ -1,61 +1,97 @@
...
+  }, []);
 
   return (
     <div className="min-h-screen flex flex-col justify-center items-center">
       <h1 className="text-2xl font-bold mb-4">PDF Viewer</h1>
       <div className="flex-row ">
-      <input type="file" id="pdfUpload" accept="application/pdf" onChange={handleFileChange}  className="mb-4 p-2 border border-gray-400 rounded"/>
-      <canvas id="pdfCanvas" className="border border-gray-300 shadow-lg"></canvas>
+        <input
+          type="file"
```
Comment (otaku-tekkurai) [inline app/page.tsx:78]: change the canvas name to proper container naming

File: app/page.tsx (hunk 2 above)
Comment (pritesh177) [reply app/page.tsx:39]: It is an format for loading pdf in pdfJs

**Pull request merged**